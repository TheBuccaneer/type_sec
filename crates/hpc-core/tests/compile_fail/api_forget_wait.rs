#![deny(unused_must_use)]
use hpc_core::*;

fn main() {
    let ctx = Context::create_context().unwrap();
    let queue = ctx.create_queue().unwrap();
    let buf = ctx
        .create_buffer::<u8>(16).unwrap()
        .enqueue_write(&queue, &[0u8; 16]).unwrap();
    let kernel = Kernel::from_source(&ctx, "kernel void dummy(__global uchar* buf) {}", "dummy").unwrap();

    // Komplettes Ignorieren – löst must_use-Fehler aus:
    buf.enqueue_kernel(&queue, &kernel, 16).unwrap();
}
