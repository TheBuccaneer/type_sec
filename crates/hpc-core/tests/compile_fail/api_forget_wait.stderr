error: unused `EventToken` in tuple element 1 that must be used
  --> tests/compile_fail/api_forget_wait.rs:13:5
   |
13 |     buf.enqueue_kernel(&queue, &kernel, 16).unwrap();
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: GPU work is in-flight: call wait(event, buf) to complete it
note: the lint level is defined here
  --> tests/compile_fail/api_forget_wait.rs:1:9
   |
1  | #![deny(unused_must_use)]
   |         ^^^^^^^^^^^^^^^
